backend/src/ee/services/dynamic-secret/providers/sql-database.ts

```diff
@@ -8,16 +8,92 @@ import { alphaNumericNanoId } from "@app/lib/nanoid";
 
 import { TGatewayServiceFactory } from "../../gateway/gateway-service";
 import { verifyHostInputValidity } from "../dynamic-secret-fns";
-import { DynamicSecretSqlDBSchema, SqlProviders, TDynamicProviderFns } from "./models";
+import { DynamicSecretSqlDBSchema, PasswordRequirements, SqlProviders, TDynamicProviderFns } from "./models";
 
 const EXTERNAL_REQUEST_TIMEOUT = 10 * 1000;
 
-const generatePassword = (provider: SqlProviders) => {
-  // oracle has limit of 48 password length
-  const size = provider === SqlProviders.Oracle ? 30 : 48;
+const DEFAULT_PASSWORD_REQUIREMENTS = {
+  minLength: 48,
+  maxLength: 48,
+  required: {
+    lowercase: 1,
+    uppercase: 1,
+    digits: 1,
+    symbols: 0
+  },
+  allowedCharacters: {
+    lowercase: 'abcdefghijklmnopqrstuvwxyz',
+    uppercase: 'ABCDEFGHIJKLMNOPQRSTUVWXYZ',
+    digits: '0123456789',
+    symbols: '-_.~!*'
```

Comment:
It might be unlikely for a user to specifically want certain uppercase characters or lowercase characters. Would it make sense to just have this as a checkbox?

like:
[ ] lowercase
[ ] uppercase
[ ] digits

and then maybe the symbols can be customized

Reply:
I think the list of allowed lowercase letters, uppercase letters, digits, and symbols would have to be treated as sensitive details because it significantly reduces the search space especially if they define a limited set.

Reply:
done